# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/svg/SVGRect.cpp
# Commit: 61a21b2c9e6e
# Full Hash: 61a21b2c9e6e9df619cf74f656d3989250939414
# Author: longsonr <longsonr@gmail.com>
# Date: 2019-08-20 09:38:33
# Description:
#   Bug 1572904 - refactor cycle collection code after DOMSVGAngle r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D42453
# ==============================================================================

diff -r b85cb0cd1e49 -r 61a21b2c9e6e dom/svg/SVGRect.cpp
--- a/dom/svg/SVGRect.cpp	Mon Aug 19 23:21:27 2019 +0000
+++ b/dom/svg/SVGRect.cpp	Mon Aug 19 19:10:03 2019 +0000
@@ -19,15 +19,10 @@
 //----------------------------------------------------------------------
 // nsISupports methods:
 
-NS_IMPL_CYCLE_COLLECTION_WRAPPERCACHE(SVGRect, mParent)
-
-NS_IMPL_CYCLE_COLLECTING_ADDREF(SVGRect)
-NS_IMPL_CYCLE_COLLECTING_RELEASE(SVGRect)
+NS_SVG_VAL_IMPL_CYCLE_COLLECTION_WRAPPERCACHED(SVGRect, mParent)
 
-NS_INTERFACE_MAP_BEGIN_CYCLE_COLLECTION(SVGRect)
-  NS_WRAPPERCACHE_INTERFACE_MAP_ENTRY
-  NS_INTERFACE_MAP_ENTRY(nsISupports)
-NS_INTERFACE_MAP_END
+NS_IMPL_CYCLE_COLLECTION_ROOT_NATIVE(SVGRect, AddRef)
+NS_IMPL_CYCLE_COLLECTION_UNROOT_NATIVE(SVGRect, Release)
 
 //----------------------------------------------------------------------
 // implementation: